prusti-contracts/src/ghost.rs

```diff
@@ -0,0 +1,139 @@
+use std::ops::*;
+use core::marker::PhantomData;
+
+pub unsafe auto trait Ghost {}
+
+macro_rules! implement_ghost_type {
+    ($ghost_type: ident) => {
+        pub struct $ghost_type;
+        unsafe impl Ghost for $ghost_type{}
+    };
+}
+
+macro_rules! implement_ghost_type_generic {
+    ($ghost_type: ident) => {
+        pub struct $ghost_type<T: Ghost> {
+            _type: PhantomData<T>
+        }
+        impl <T: Ghost> $ghost_type<T> {
+            ///  let seq_inst: GhostSeq<GhostInt> = GhostSeq::new();
```

Comment:
Code examples should be inside Markdown code blocks:
```suggestion
            ///  ```rust
            ///  let seq_inst: GhostSeq<GhostInt> = GhostSeq::new();
            ///  ```
```